### PR TITLE
Keep named arguments during eta expansion

### LINKED ISSUE #1096: EtaExpansion.liftArgs removes NamedArgs
This could make correlating trees with positions harder.
We have an agreement that NamedArgs should survive until FirstTransform.

The place where it's eliminated:
https://github.com/lampepfl/dotty/blob/master/src/dotty/tools/dotc/typer/EtaExpansion.scala#L53

### PR DESCRIPTION
**Pull request by odersky**
Fixes #1096. Review by @xeno-by

### CODE REVIEW

**Comment by xeno-by** [comment]
Does this `NamedArg` carry a reference to the corresponding method parameter?

**Comment by DarkDimius** [comment]
the `name` field indicates what was the name of method parameter.

File: src/dotty/tools/dotc/typer/EtaExpansion.scala [hunk 1]
```diff
@@ -48,13 +48,11 @@ object EtaExpansion {
   }
 
   /** Lift a function argument, stripping any NamedArg wrapper */
```
Comment (smarter) [inline src/dotty/tools/dotc/typer/EtaExpansion.scala]: I think the documentation needs to be updated, since we don't strip the NamedArg anymore.

**Comment by xeno-by** [comment]
@DarkDimius Is there a stronger connection, e.g. via a symbol?

**Comment by DarkDimius** [comment]
No. Symbols of arguments of methods are only accessible inside the body of the method in Dotty.

File: src/dotty/tools/dotc/typer/EtaExpansion.scala (hunk 1 above)
Comment (odersky) [inline src/dotty/tools/dotc/typer/EtaExpansion.scala]: We do strip it, but then rewrap it around something else. I meant to do the change, but then decided it's too much busywork for too minor an improvement.

**Pull request merged**